include/silo/zstdfasta/zstdfasta_table_reader.h

```diff
@@ -42,9 +44,9 @@ class ZstdFastaTableReader {
 
    std::optional<std::string> nextSkipGenome();
 
-   std::optional<std::string> next(std::string& genome);
+   std::optional<std::string> next(std::optional<std::string>& genome);
```

Comment:
Hmm I am not sure whether it is just me, but I find it a bit confusing, that we now have two optionals here, that refer to different things. But I also did not find a way around that that is elegant.
Of course the idiomatic thing to do, would be to return a `std::optional` of a custom data type, that does now have a key and an optional string. But this was initially not chosen, as it can be often much more efficient (especially for long strings), to write them directly into a preallocated string reference, given as parameter.

Also I am not sure, whether there the performance penalty is already paid by writing into a `std::optional<std::string>`

Reply:
Would be happy to merge it like this, though. What do you think?

Reply:
As discussed, we leave it as it is for now and maybe investigate again later whether we have performance issues.